src/views/Tournament/Tournament.tsx

```diff
@@ -184,53 +185,109 @@ export function Tournament(): JSX.Element {
         lead_time_seconds: 1800,
         base_points: 10.0,
     });
-    const [rounds, setRounds] = React.useState<any[]>([]);
+
     const [editing, setEditing] = React.useState(tournament_id === 0);
     const [raw_rounds, setRawRounds] = React.useState<any[]>([]);
-    const [selected_round_idx, setSelectedRoundIdx] = React.useState<
-        number | "standings" | "roster"
-    >(0);
-    const [sorted_players, setSortedPlayers] = React.useState<any[]>([]);
-    const [players, setPlayers] = React.useState<{ [id: string]: TournamentPlayer }>({});
-    const [is_joined, setIsJoined] = React.useState(false);
+    const [clicked_round_idx, setClickedRoundIdx] = React.useState<
```

Comment:
Okay. This one's a bit weird, I admit, and there's a behaviour change (which I think is mostly good, but maybe it's debatable).

Previously there was `selected_round_idx` as state and a forwarding function that was used for setting it:
```
const [selected_round_idx, setSelectedRoundIdx] = React.useState(...);
const setSelectedRound = (idx: number) => setSelectedRoundIdx(idx);
```
(I left behind this forwarding function, it just points to `setClickedRoundIdx` now. Maybe shouldn't have?)

The reason I've added an extra layer, "clicked", is to distinguish between:
- "is this selected because the user clicked on it?"
- "is this selected because we don't have any rounds loaded yet?"

Say a user opens a tournament that's currently in round 2 of 5. Here is the behaviour we we'll get right now:
1. User opens the tournament. No rounds are displayed (but `selected_round_idx` will be 0).
2. Once the rounds load, round 2 (active round) is selected automatically.
3. Round 2 finishes and round 3 starts. User automatically switched to round 3 (not having made a manual selection by clicking).
4. User clicks on round 1, which selects round 1.
5. Round 3 finishes and round 4 starts. User still has round 1 selected (having clicked on it).
6. User clicks on round 4, which selects round 4.
7. Round 4 finishes and round 5 starts. User still has round 4 selected (having clicked on it).
8. User switches to a different tournament that's in round 1 via OGS search navigation. User gets reverted to (1).

Comments:
- (2) is important. It used to be accomplished by calling `setSelectedRoundIdx` any time rounds were loaded. Now the "default" round selection is just part of the assignment of `selected_round_idx`, and loading rounds doesn't change the state of what the user clicked on.
- (3) is a behaviour I'm not sure if we want, but it's not a change. Comes naturally from (2). ("User can follow along a tournament as it's going")
- (5) is new behaviour. I think we want it. Accomplished by having `clicked_round_idx` start out as `null` (did they pick this round? or are they just following along?).
- (7) is new behaviour. Maybe we don't want it. Maybe, instead, if they manually click on the active round, we should revert back to (3) by something like:
    ```
    const setSelectedRound = (idx: number) => {
        setClickedRoundIdx(idx == default_round_idx ? null : idx);
    };
    ```

WDYT? Given that walkthrough, any other thoughts on naming? (Would this be clearer/easier to follow if we introduced a `default_round_idx` computed value?)

Reply:
Made some changes: https://github.com/online-go/online-go.com/compare/af4783d13b5abc16989920482ae8678a0877bdc8...856dd687a3f48e4cf6871b85e9e7cdefe9061bb9. I think the code is more clear now, and it takes the change I suggested in my comment about (7) above.

Let me know what you think!

Reply:
Oh cool, thanks for the explanation!

I think the behavior you specified sounds reasonable, and "explicitly selected" is a good name for that variable according to this description 🙂